core/src/Instance/AadAuthority.cs

```diff
@@ -118,8 +125,11 @@ protected override string GetDefaultOpenIdConfigurationEndpoint()
 
         internal static bool IsInTrustedHostList(string host)
         {
-            return !string.IsNullOrEmpty(
-                       TrustedHostList.FirstOrDefault(a => string.Compare(host, a, StringComparison.OrdinalIgnoreCase) == 0));
+            bool isInList = 
+                !string.IsNullOrEmpty(
+                    TrustedHostList.FirstOrDefault(a => string.Compare(host, a, StringComparison.OrdinalIgnoreCase) == 0));
+            isInList |= host.EndsWith(B2CTrustedHost);
```

Comment:
>isInList |= host.EndsWith(B2CTrustedHost); [](start = 12, length = 42)

should be moved to the B2CAuthority... could also make the additional check in the line above, also what about other custom domains... not sure if it will work or not.